Owncloud iOs Client/Network/Uploads/PrepareFilesToUpload.m

```diff
@@ -145,117 +138,95 @@ - (void)uploadFileFromGallery:(PHAsset *)assetToUpload andRemoteFolder:(NSString
-            dateFormatter = [[NSDateFormatter alloc] init];
-            [dateFormatter setDateFormat:@"yyyy-MM-dd-HH-mm-ss.SSS"];
-            dateFormatter.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"];
-        });
-        
-        NSURL *url = info[@"PHImageFileURLKey"];
-        NSString *fileExtension = url.pathExtension;
-        
-        //Use a temporary name with a date identification
-        NSString *temporaryFileName = [NSString stringWithFormat:@"IMG_%@.%@", [dateFormatter stringFromDate:[NSDate date]], fileExtension];
-        NSString *localPath = [[UtilsUrls getTempFolderForUploadFiles] stringByAppendingPathComponent:temporaryFileName];
-        
-        NSFileManager *fileManager = [NSFileManager defaultManager];
-        if ([fileManager fileExistsAtPath:localPath]) {
-            [fileManager removeItemAtPath:localPath error:nil];
-        }
-        
-        [fileManager createFileAtPath:localPath contents:imageData attributes:nil];
-        
-        UploadsOfflineDto *currentUpload = [[UploadsOfflineDto alloc] init];
-        currentUpload.originPath = localPath;
-        currentUpload.destinyFolder = remoteFolder;
-        currentUpload.uploadFileName = temporaryFileName;
-        currentUpload.estimateLength = imageData.length;;
-        currentUpload.userId = currentUser.idUser;
-        currentUpload.isLastUploadFileOfThisArray = isLastUploadFileOfThisArray;
-        currentUpload.status = waitingAddToUploadList;
-        currentUpload.chunksLength = k_lenght_chunk;
-        currentUpload.uploadedDate = 0;
-        currentUpload.kindOfError = notAnError;
-        currentUpload.isInternalUpload = YES;
-        currentUpload.taskIdentifier = 0;
-        
-        [self.listOfUploadOfflineToGenerateSQL addObject:currentUpload];
-        
-        if([self.listOfAssetsToUpload count] > 0) {
-            //We have more files to process
-            [self startWithTheNextAsset];
-        } else {
+    if (assetToUpload.mediaType == PHAssetMediaTypeImage) {
+        [[PHImageManager defaultManager] requestImageDataForAsset:assetToUpload options:nil resultHandler:^(NSData * _Nullable imageData, NSString * _Nullable dataUTI, UIImageOrientation orientation, NSDictionary * _Nullable info) {
             
-            //We finish all the files of this block
-            DLog(@"self.listOfUploadOfflineToGenerateSQL: %lu", (unsigned long)[self.listOfUploadOfflineToGenerateSQL count]);
+            NSURL *url = info[@"PHImageFileURLKey"];
+            NSString *fileExtension = url.pathExtension;
             
-            //In this point we have all the files to upload in the Array
-            [ManageUploadsDB insertManyUploadsOffline:self.listOfUploadOfflineToGenerateSQL];
+            //Use a temporary name with a date identification
+            NSString *temporaryFileName = [NSString stringWithFormat:@"IMG_%@.%@", [dateFormatter stringFromDate:[NSDate date]], fileExtension];
```

Comment:
We include here a regression bug with the refactor of instant uploads. Please keep using same name as before. For the new PHAssets a new method was created, you can use it [getComposeNameFromPHAsset](https://github.com/owncloud/ios/blob/master/Owncloud%20iOs%20Client/Utils/FileNameUtils.m#L467)